frostsnap_core/src/device.rs

```diff
@@ -490,27 +502,44 @@ impl FrostSigner {
         }
     }
 
-    pub fn action_state_name(&self) -> &'static str {
-        self.action_state
-            .as_ref()
-            .map(|x| x.name())
-            .unwrap_or("None")
-    }
-
-    pub fn key_backup(&mut self, key_id: KeyId) -> Result<String, ActionError> {
-        match self.keys.get(&key_id) {
-            Some(key) => {
+    pub fn display_backup_ack(&mut self) -> Result<Vec<DeviceSend>, ActionError> {
+        match self.action_state {
+            Some(SignerState::DisplayBackup {
+                key_id,
+                awaiting_ack: true,
+            }) => {
+                let key = self.keys.get(&key_id).expect("key must exist");
```

Comment:
Yep we now error earlier, in `recv_coordinator_message`, if a coordinator requests a backup of an unknown key